### ISSUE TITLE
Error code not being returned

### ISSUE THREAD

**Issue by rickharrison** [raycmorgan/sqlbox#12]
This may actually be working as intended, but I think based on the docs, this error should be returning a 409. Instead, it is returning the postgres error code of 23505 for a unique index violation. When saving my model, the error I am getting is as follows:

```
{ [error: duplicate key value violates unique constraint "landlords_email_key"]
  length: 175,
  name: 'error',
  severity: 'ERROR',
  code: '23505',
  detail: 'Key (email)=(rick@ninerobot.com) already exists.',
  hint: undefined,
  position: undefined,
  internalPosition: undefined,
  internalQuery: undefined,
  where: undefined,
  file: 'nbtinsert.c',
  line: '396',
  routine: '_bt_check_unique' }
```

This isn't an issue for me as I can just check for 23505, but I wanted to let you know in case you think this should actually be outputting a 409.

**Comment by raycmorgan** [comment]
You are correct. The `save` method should convert the error code to a 409 and also have a proper message containing what happened. Also, I think the error should have a property for which key failed uniqueness, probably in the same format of a validation error.

**Comment by raycmorgan** [comment]
Just pushed the fix for this in npm version 0.2.7.

**Comment by rickharrison** [comment]
Awesome, thanks. I'll pull it down and test it out tonight.
